observability/src/main/java/io/grpc/observability/logging/CloudLoggingHandler.java

```diff
@@ -0,0 +1,190 @@
+    this(level, null, null);
+  }
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging.
+   *
+   * @param level set the level for which message levels will be logged by the custom logger
+   * @param logName the name of the log to which log entries are written
+   */
+  public CloudLoggingHandler(Level level, String logName) {
+    this(level, logName, null);
+  }
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging.
+   *
+   * @param level set the level for which message levels will be logged by the custom logger
+   * @param logName the name of the log to which log entries are written
+   * @param overrideProjectId the value of cloud project id to which logs are sent to by the custom
+   *     logger
+   */
+  public CloudLoggingHandler(Level level, String logName, String overrideProjectId) {
+    baseLevel = level.equals(DEFAULT_LOG_LEVEL) ? Level.FINEST : level;
+    setLevel(baseLevel);
+
+    cloudLogName = logName != null ? logName : DEFAULT_LOG_NAME;
+
+    try {
+      // TODO(dnvindhya) read the value from config instead of taking it as an argument
+      String projectId = overrideProjectId != null ? overrideProjectId : null;
+
+      if (!Strings.isNullOrEmpty(projectId)) {
+        loggingOptions = LoggingOptions.newBuilder().setProjectId(overrideProjectId).build();
+      } else {
+        loggingOptions = LoggingOptions.getDefaultInstance();
+      }
+
+      loggingClient = loggingOptions.getService();
+    } catch (Exception ex) {
+      reportError("Error in initializing Cloud Logging", ex, ErrorManager.OPEN_FAILURE);
+      throw ex;
+    }
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if (!(record instanceof LogRecordExtension)) {
+      throw new IllegalArgumentException("Expected record of type LogRecordExtension");
+    }
+
+    Level logLevel = record.getLevel();
+    GrpcLogRecord protoRecord = ((LogRecordExtension) record).getGrpcLogRecord();
+
+    writeLog(protoRecord, logLevel);
+  }
+
+  private void writeLog(GrpcLogRecord logProto, Level logLevel) {
+    try {
+      Severity cloudLogLevel = getCloudLoggingLevel(logLevel);
+      Map<String, Object> mapPayload = protoToMapConverter(logProto);
+
+      LogEntry grpcLogEntry =
+          LogEntry.newBuilder(JsonPayload.of(mapPayload))
+              .setSeverity(cloudLogLevel)
+              .setLogName(cloudLogName)
+              .setResource(MonitoredResource.newBuilder("global").build())
+              .build();
+
+      loggingClient.write(Collections.singleton(grpcLogEntry));
+    } catch (Exception ex) {
+      getErrorManager().error("Error in writing logs to cloud logging", ex,
+          ErrorManager.WRITE_FAILURE);
+    }
+  }
+
+  /* Google Cloud Logging API only accepts String payload or Json Payload
+  Adding all the grpc o11y logging values as string makes it difficult to look up or index
+  Hence we are going with the Json Payload approach
```

Comment:
this was part of our design decision so I think comment is not needed here. Also we have not addressed the issue of ints appearing as double (e.g. `3.0` instead of 3)?